### PR TITLE
Tighted pod security and adjust annotations

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Several changes to how k8s pods are created:
- security is locked down by default (eg allow privilege escalation = false, run as root = false etc)
- container level security context supported
- set seccomp and apparmour profiles
- move model and controller uuid and resource tags to annotations
- rename juju-application to juju-app

Also drivebt fix for wired tiger cache size.

## QA steps

bootstrap k8s
deploy kubeflow

### CODE REVIEW

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -2152,16 +2181,20 @@ func operatorPod(podName, appName, agentPath, operatorImagePath, version string,
...
 	appTag := names.NewApplicationTag(appName)
-	podLabels := make(map[string]string)
+	podAnnotations := make(map[string]string)
 	for k, v := range tags {
-		podLabels[k] = v
+		podAnnotations[k] = v
 	}
-	podLabels[labelVersion] = version
+	podAnnotations[labelVersion] = version
+	podAnnotations["apparmor.security.beta.kubernetes.io/pod"] = "runtime/default"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/k8s.go]: do we wanna use `podAnnotations` here as well?

File: mongo/service.go [hunk 2]
```diff
@@ -459,10 +459,14 @@ func generateConfig(mongoPath string, dataDir string, statePort int, oplogSizeMB
 		//BindIP:                "127.0.0.1", // TODO(tsm): use machine's actual IP address via dialInfo
 	}
 
-	if usingMongo34orAbove && useLowMemory && usingWiredTiger {
-		mongoArgs.WiredTigerCacheSizeGB = Mongo34LowCacheSize
-	} else if usingWiredTiger {
-		mongoArgs.WiredTigerCacheSizeGB = LowCacheSize
+	if usingWiredTiger {
```
Comment (ycliuhw) [inline mongo/service.go]:
```
if usingWiredTiger && useLowMemory {

}
```
then less one {}?

File: caas/kubernetes/provider/k8s.go (hunk 1 above)
Comment (wallyworld) [reply caas/kubernetes/provider/k8s.go]: yup

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**